src/settings/SettingsStore.ts

```diff
@@ -61,20 +63,39 @@ for (const key of Object.keys(LEVEL_HANDLERS)) {
 }
 
 const LEVEL_ORDER = [
-    'device', 'room-device', 'room-account', 'account', 'room', 'config', 'default',
+    SettingLevel.DEVICE,
+    SettingLevel.ROOM_DEVICE,
+    SettingLevel.ROOM_ACCOUNT,
+    SettingLevel.ACCOUNT,
+    SettingLevel.ROOM,
+    SettingLevel.CONFIG,
+    SettingLevel.DEFAULT,
 ];
 
+export type CallbackFn = (
+    settingName: string,
+    roomId: string,
+    atLevel: SettingLevel,
+    newValAtLevel: any,
+    newVal: any,
+) => void;
+
+interface HandlerMap {
+    // @ts-ignore - TS wants this to be a string key but we know better
+    [level: SettingLevel]: SettingsHandler;
+}
```

Comment:
but its called a Map :'(

Reply:
refactoring all of settings to use maps is a different PR's problem.